### PR TITLE
Allow :prefix option on metrics creation

### PR DESCRIPTION
**Pull request by timothysmith0609**
Closes #69 

Allows clients to pass in `:prefix` as an option to a metric. This allows classes that extend StatsD to avoid polluting the global namespace. The motivation for this is [KubernetesDeploy #384](https://github.com/Shopify/kubernetes-deploy/pull/384), though it seems it's not the first app to feel the need for this.

The change is backwards compatible and purely opt-in

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ require 'bundler/gem_tasks'
 require 'rake/testtask'
 
 Rake::TestTask.new('test') do |t|
-  t.ruby_opts << '-rubygems'
+  t.ruby_opts << '-r rubygems'
```
Comment (karanthukral) [inline Rakefile]: unrelated change>

File: lib/statsd/instrument/metric.rb [hunk 2]
```diff
@@ -47,7 +49,11 @@ def initialize(options = {})
     @type = options[:type] or raise ArgumentError, "Metric :type is required."
     @name = options[:name] or raise ArgumentError, "Metric :name is required."
     @name = normalize_name(@name)
-    @name = StatsD.prefix ? "#{StatsD.prefix}.#{@name}" : @name unless options[:no_prefix]
+    @name = if options[:prefix]
```
[the remark this answers is not in the document]
Comment (timothysmith0609) [reply lib/statsd/instrument/metric.rb]: `prefix` will override (which I should document).
I could be persuaded to have `:no_prefix` have greater precedence but it's 6 of 1, half-dozen of the other.

File: Rakefile (hunk 1 above)
Comment (timothysmith0609) [reply Rakefile]: CI was complaining about this (and was in fact preventing me from testing locally)

File: lib/statsd/instrument.rb [hunk 3]
```diff
@@ -448,11 +448,16 @@ def parse_options(value, metric_options)
...
   def collect_metric(type, name, value, metric_options)
     value, metric_options = parse_options(value, metric_options)
-
-    options = hash_argument(metric_options).merge(type: type, name: name, value: value)
+    metric_options = hash_argument(metric_options)
+    options = metric_options.merge(
+      type: type,
+      name: name,
+      value: value,
+      prefix: metric_options[:prefix]
```
Comment (fw42) [inline lib/statsd/instrument.rb]: what does this line actually do? I'm reading this as `metric_options.merge(prefix: metric_options[:prefix])`, which doesn't look like it's doing anything, is it?

File: lib/statsd/instrument/metric.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (fw42) [reply lib/statsd/instrument/metric.rb]: Hm. `no_prefix` having precedence sounds less surprising to me.

File: lib/statsd/instrument.rb (hunk 3 above)
Comment (timothysmith0609) [reply lib/statsd/instrument.rb]: Ah, you are right I goofed
